include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -0,0 +1,310 @@
+            callback(alignment_result_type{res});
+        }
+    }
+    //!\}
+
+protected:
+    /*!\brief Checks wether the band given the sequence sizes is valid
+     *
+     * \param[in] sequence1_size The size of the first sequence.
+     * \param[in] sequence2_size The size of the second sequence.
+     *
+     * \throws seqan3::invalid_alignment_configuration if the band is invalid for the given sequence sizes and the
+     *         alignment configuration.
+     */
+    void check_valid_band_configuration(size_t const sequence1_size, size_t const sequence2_size) const
+    {
+        bool const upper_diagonal_ends_before_last_cell = (upper_diagonal + sequence2_size) < sequence1_size;
+        bool const lower_diagonal_ends_behind_last_cell = (-lower_diagonal + sequence1_size) < sequence2_size;
+
+        if (upper_diagonal_ends_before_last_cell || lower_diagonal_ends_behind_last_cell)
+            throw invalid_alignment_configuration{to_string("The selected band [",
+                                                            lower_diagonal,
+                                                            ":",
+                                                            upper_diagonal,
+                                                            "] does not cover the last cell.")};
+    }
+
+    /*!\brief Compute the actual banded alignment.
+     * \tparam sequence1_t The type of the first sequence; must model std::ranges::forward_range.
+     * \tparam sequence2_t The type of the second sequence; must model std::ranges::forward_range.
+     *
+     * \param[in] sequence1 The first sequence to compute the alignment for.
+     * \param[in] sequence2 The second sequence to compute the alignment for.
+     *
+     * \details
+     *
+     * In the banded alignment the iteration of the inner columns is split into two phases. The first phase
+     * reuses the unbanded column computation and assumes that the banded score matrix always starts at the beginning
+     * of the matrix. In the second phase the special interface
+     * seqan3::detail::pairwise_alignment_algorithm_banded::compute_band_column is used to compute the banded column.
+     * The current implementation assumes that the first cell of the current score matrix column is always the first
+     * cell to compute in every column. The respective seqan3::detail::score_matrix_single_column is resized to the band
+     * size plus one additional field to cover the end of the band where the end of the column is not reached yet.
+     * This cell will be never written to but only read from, i.e. it represents minus infinity. This allows the
+     * algorithm to reuse the standard unbanded cell computation.
+     *
+     * The coordinate matrix represents the global matrix index and not the local band coordinate. Data structures that
+     * require the coordinate might need to map the global matrix coordinate to their local coordinate.
```

Comment:
I don't understand what you have in mind here. Which data structures do you mean?

Reply:
better now?